File: services/gateway/app/models/veiculo.py
import uuid
from datetime import datetime

from sqlalchemy import Column, Date, DateTime, ForeignKey, Integer, Numeric, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.db.session import Base


class Veiculo(Base):
    __tablename__ = "veiculos"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    cliente_id = Column(UUID(as_uuid=True), ForeignKey("clientes.id", ondelete="CASCADE"),
                        nullable=False, index=True)

    modelo = Column(String(80), nullable=False)
    versao = Column(String(80), nullable=False)
    ano = Column(Integer, nullable=False)
    vin = Column(String(17), unique=True, nullable=False, index=True)
    placa = Column(String(8), nullable=True, index=True)
    data_compra = Column(Date, nullable=False)
    valor_compra = Column(Numeric(12, 2), nullable=False)
    concessionaria_id = Column(String(40), nullable=False, index=True)
    criado_em = Column(DateTime(timezone=True), default=datetime.utcnow, nullable=False)

    cliente = relationship("Cliente", back_populates="veiculos")
    leads = relationship("Lead", back_populates="veiculo")

File: services/core/src/main/java/com/previopls/entity/Veiculo.java
package com.previopls.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "veiculos", indexes = {
        @Index(name = "ix_veiculos_cliente_id", columnList = "cliente_id"),
        @Index(name = "ix_veiculos_vin", columnList = "vin", unique = true),
        @Index(name = "ix_veiculos_concessionaria_id", columnList = "concessionaria_id")
})
@Getter
@Setter
@NoArgsConstructor
public class Veiculo {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "uuid")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @Column(nullable = false, length = 80)
    private String modelo;

    @Column(nullable = false, length = 80)
    private String versao;

    @Column(nullable = false)
    private Integer ano;

    @Column(nullable = false, unique = true, length = 17)
    private String vin;

    @Column(name = "data_compra", nullable = false)
    private LocalDate dataCompra;

    @Column(name = "valor_compra", nullable = false, precision = 12, scale = 2)
    private BigDecimal valorCompra;

    @Column(name = "concessionaria_id", nullable = false, length = 40)
    private String concessionariaId;

    @Column(name = "criado_em", nullable = false, updatable = false)
    private Instant criadoEm = Instant.now();
}
